### PR TITLE
Block editor: remove root appender

### PR DESCRIPTION
**Pull request by ellatrix**
## What?

Let's see what tests fail and how this tests manually.

* In the site editor, this appender is really weird: I don't expect to insert any blocks _after_ the site footer.
* In the post editor, we always bottom pad the editor, so we should use that space to allow people to insert blocks, by adding a paragraph on click. You can also press Enter from the last block, or use the dropdown menu.

If this isn't working for the post editor, I think this is a writing flow issue that we need to resolve differently.

## Why?

* The editor should be a clear preview of the front-end, without much UI floating around for unselected blocks.
* Reduce layout shifts.
* Improve performance (this appender actually triggers a request for reusable blocks when loading the editor).

### CODE REVIEW

**Comment by jasmussen** [comment]
This is working well for me especially in the site editor where shown in this GIF is the appender causing layout shifts at the end of each document, including template parts edited in isolation mode:

trunk

This branch fixes that by omitting those appenders causing the shifts, as shown in this GIF:

branch

One nuance is on the post editor, there's currently a heuristic to show this appender after the last block, if that block is not a text block. Shown here, the appender after an empty image, from trunk:

trunk post

Here, the same case, shown without said appender:
branch post

I think it's fine to remove it there, you click the image and press Enter. But you can't click the space below, to type text.

**Comment by jasmussen** [comment]
By the way those two GIFs also show an issue that appears to be a regression in trunk: if you go to the site editor, edit a template part in isolation (focus mode), then exit out to site view and enter the site editor again, the site editor itself will be in focus mode. It seems like the query that gets appended when you enter template part editing mode (`?focusMode=true`) does not get properly cleared when exiting focus editing mode.

**Comment by ellatrix** [comment]
@jasmussen I just pushed a commit to make the 40vh space below the blocks append a default block on click.

**Comment by jasmussen** [comment]
That works great! Can that have any side effects in the numerous other editing contexts we have?

**Comment by ellatrix** [comment]
I added it only for the normal post editing experience (where the typewriter is also active). This is not added for site editor post editing, where we also don't have the typewriter padding. I guess at some point we'll have to see if we can bring this experience to the post editor inside the site editor as well. In those cases we rely solely on Enter or the dropdown menu to append a block, which I think is also fine. Maybe one little thing we _could_ do is show the appender when the last block is selected? But that would re-introduce the layout shift when clicking around.

**Comment by jasmussen** [comment]
Maybe we should start with this and see how that feels? It's between releases.

**Review by jasmussen — APPROVED** [review]
Tentative green check, but would appreciate a gut check from another.

**Comment by ellatrix** [comment]
But it's worth noting that in the site editor, nothing is really changing from trunk, because the appender there gets added to the template root, not the post-content block. The post content block does have an appender, but only if you select the pos content block itself.

**Review by richtabor — APPROVED** [review]
I like it; works as expected.

**Comment by t-hamano** [comment]
Removing the root appender may have affected accessibility. See #61391.

**Comment by alexstine** [comment]
@ellatrix and @jasmussen Let's say you insert 3 paragraph blocks and 1 image block for a total of 4 blocks. If I want to now add a 5th block, I can't do that because the image block is not a text block. Let's think from the perspective of a user with no vision, how has this experience been made better for them? I have real concerns with PRs like this one that touch lots of visual concepts and nothing is flagged for accessibility review.

If you can give me a valid explanation as to how keyboard only users are supposed to be able to always add a block from the end of the content, I'll likely just go away. What I'm not going to accept is enter will do it in some blocks but not all. Accessibility is made possible by patterns, not guesses based on which type of UI you are in.

Thanks.

**Comment by ellatrix** [comment]
> What I'm not going to accept is enter will do it in some blocks but not all.

Enter should work from the block wrapper for all blocks. Is there one where this doesn't work?

**Comment by alexstine** [comment]
@ellatrix It's another one of those hidden patterns we're expecting users to figure out. For example, the paragraph block is the block wrapper so Enter key will add a new block if the caret is near the end. Yet another concept users must understand. If in an image block where focus is placed on the first tabbable by default, users would have to very specifically know to press Left Arrow then Enter to add a new block. I think at a certain level, this is fine for users like you or myself who understand how the editor works. For people who are simply trying to write a blog post with some different blocks, this could be the worst user experience. Even navigation mode made this a fairly hidden feature but I do not like to think about what happens if we remove patterns users have already discovered. The only other real discoverable method to add a block is via the add block button, right? I think you can get to it with a double Shift+Tab?

File: packages/edit-post/src/components/visual-editor/index.js [hunk 1]
```diff
@@ -53,6 +54,8 @@ export default function VisualEditor( { styles } ) {
 		[]
 	);
 
+	const paddingAppenderRef = usePaddingAppender();
```
Comment (youknowriad) [inline packages/edit-post/src/components/visual-editor/index.js:57]: Should we move this one (and the padding bottom) to the "editor" package (VisualEditor component) instead? That way the same behavior would apply to pages (in non preview mode) in the site editor too.

**Comment by Mamaduka** [comment]
Here's a small follow-up for this PR - #68951.

**Pull request merged**